### PR TITLE
Fix #630 : Oppia Terminology Wiki

### LINKED ISSUE #630: Wiki: Oppia Terminology Wiki
Wiki page which describes following words, there usage in the app and how they are inter-related.

1. Topic
2. Subtopic
3. Story
4. PromotedStory
5. Chapter
6. Skill
7. Exploration
8. State
9. ConceptCard
10. ReviewCard
11. Question/QuestionPlayer

### PR DESCRIPTION
**Pull request by MohitGupta121**
## Explanation

Fix #630 : Oppia Terminology Wiki

I have introduced new terminology in the Oppia wiki file to clarify the meaning of commonly used terms such as Topics, Story, Promoted Story, Skill, Chapter, Exploration, Question, and Concept card within the Oppia and Oppia-Android codebase. Additionally, I have included a visual flowchart that illustrates the interrelationships among these terms.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

**Review by MohitGupta121 — COMMENTED** [review]
@adhiamboperes I apologize for delay PR. PTAL, Thanks!

File: wiki/Terminology in Oppia.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+S2 --> C3("Chapter-3")
+S2 --> C4("Chapter-4")
+C3 --> c4("Card-4")
+C4 --> c5("Card-5")
+C4 --> c6("Card-6")
+```
+
+## Key terms
+
+1. **Topic**: A topic is a broad term that refers to the subject content being taught (e.g. Addition/Subtraction). It's appear on Home screen of app.
```
Comment (adhiamboperes) [inline wiki/Terminology in Oppia.md]:
```suggestion
1. **Topic**: A topic is a broad term that refers to the subject content being taught (e.g. Addition/Subtraction). It appears on Home screen of app.
```

File: wiki/Terminology in Oppia.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+2. **Story**: Stories are situations/scenarios that are meant to help users understand the topic. For example, if the topic is addition, then one of the stories could be about where a kid goes to a shop and buys 3 pens and 4 pencils. It's appear when you click any "Topic" in "Lessons" tab.
```
Comment (adhiamboperes) [inline wiki/Terminology in Oppia.md]:
```suggestion
2. **Story**: Stories are situations/scenarios that are meant to help users understand the topic. For example, if the topic is addition, then one of the stories could be about where a kid goes to a shop and buys 3 pens and 4 pencils. A list of stories is shown in the "Lessons" tab when you open a topic from the home screen.
```

File: wiki/Terminology in Oppia.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+3. **Chapter**: A chapter corresponds to different lessons through which we aim to teach the topic to the users. Each chapter has an associated exploration, and this exploration may have several cards.
+4. **Promoted Story**: Promoted Story is mainly the recent Story/Chapter you played. It's shown on "home screen" with heading text "Stories For You".
```
Comment (adhiamboperes) [inline wiki/Terminology in Oppia.md]:
```suggestion
4. **Promoted Story**: Promoted Story is mainly the recent Story/Chapter you played. It is shown on the "home screen" with heading text "Stories For You".
```

File: wiki/Terminology in Oppia.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+5. **Skill**: This is a concrete learning outcome that describes something that a learner should be able to do. It is usually stated in the form “Given X, compute/calculate/draw/etc. Y.” For example: “Given a fraction, identify its numerator.”
+6. **Exploration**: This is a structured learning experience that is part of a story, and provides the learner with an active way to learn new concepts, as well as targeted feedback. It is the core unit of learning in Oppia. The flow/screens appears when any story is started is known as Exploration.
```
Comment (adhiamboperes) [inline wiki/Terminology in Oppia.md]: Isn't a chapter and exploration the same? You can combine the two definitions.

File: wiki/Terminology in Oppia.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+7. **Concept Card**: This is a non-story-based explanation of how to perform a particular skill. It serves as a reference/reminder for students who may have encountered the skill before but forgotten how to carry it out. These can access from "Revision" tab or they linked within the chapter you playing.
```
Comment (adhiamboperes) [inline wiki/Terminology in Oppia.md]:
```suggestion
7. **Concept Card**: This is a non-story-based explanation of how to perform a particular skill. It serves as a reference/reminder for students who may have encountered the skill before but forgotten how to carry it out. These can access from "Revision" tab or are linked within the chapter you are playing. 
```

File: wiki/Terminology in Oppia.md
Comment (adhiamboperes) [inline wiki/Terminology in Oppia.md:1]: File name needs to be hyphenated, it is causing a parse exception. Please see your CI results

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @MohitGupta121! Please review the suggestions and also update your PR description to add a bit for information.

**Comment by MohitGupta121** [comment]
Okay, Thanks @adhiamboperes I'll look into these changes.

File: wiki/Terminology in Oppia.md
Comment (MohitGupta121) [reply wiki/Terminology in Oppia.md:1]: Thanks, I missed that. Done.

File: wiki/Terminology in Oppia.md (hunk 1 above)
Comment (MohitGupta121) [reply wiki/Terminology in Oppia.md]: Done.

File: wiki/Terminology in Oppia.md (hunk 2 above)
Comment (MohitGupta121) [reply wiki/Terminology in Oppia.md]: Done.

File: wiki/Terminology in Oppia.md (hunk 3 above)
Comment (MohitGupta121) [reply wiki/Terminology in Oppia.md]: Done.

File: wiki/Terminology in Oppia.md (hunk 4 above)
Comment (MohitGupta121) [reply wiki/Terminology in Oppia.md]: Yes, Right. Thanks!
Updated.

File: wiki/Terminology in Oppia.md (hunk 5 above)
Comment (MohitGupta121) [reply wiki/Terminology in Oppia.md]: Done.

**Review by MohitGupta121 — COMMENTED** [review]
@adhiamboperes PTAL, updated PR description with explanation, and suggested changes. Thanks!

File: wiki/Terminology-in-Oppia.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+S2 --> C3("Chapter-3")
+S2 --> C4("Chapter-4")
+C3 --> c4("Card-4")
+C4 --> c5("Card-5")
+C4 --> c6("Card-6")
+```
+
+## Key terms
+
+1. **Topic**: A topic is a broad term that refers to the subject content being taught (e.g. Addition/Subtraction). It appears on Home screen of app.
```
Comment (adhiamboperes) [inline wiki/Terminology-in-Oppia.md]:
```suggestion
1. **Topic**: A topic is a broad term that refers to the subject content being taught (e.g. Addition/Subtraction). A list of available topics appears on the Home screen of app.
```

File: wiki/Terminology-in-Oppia.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+2. **Story**: Stories are situations/scenarios that are meant to help users understand the topic. For example, if the topic is addition, then one of the stories could be about where a kid goes to a shop and buys 3 pens and 4 pencils. A list of stories is shown in the "Lessons" tab when you open a topic from the home screen.
```
Comment (adhiamboperes) [inline wiki/Terminology-in-Oppia.md]:
```suggestion
2. **Story**: Stories are situations/scenarios that are meant to help users understand the topic. For example, if the topic is addition, then one of the stories could be about a kid going to a shop and to buy 3 pens and 4 pencils. A list of stories is shown in the "Lessons" tab when you open a topic from the home screen.
```

File: wiki/Terminology-in-Oppia.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+3. **Promoted Story**: Promoted Story is mainly the recent Story/Chapter you played. It is shown on the "home screen" with heading text "Stories For You".
+4. **Skill**: This is a concrete learning outcome that describes something that a learner should be able to do. It is usually stated in the form “Given X, compute/calculate/draw/etc. Y.” For example: “Given a fraction, identify its numerator.”
+5. **Exploration/Chapter**: This is a structured learning experience that is part of a story, and provides the learner with an active way to learn new concepts, as well as targeted feedback. It is the core unit of learning in Oppia. The flow/screens appears when any story is started is known as Exploration/Chapter.
```
Comment (adhiamboperes) [inline wiki/Terminology-in-Oppia.md]:
```suggestion
5. **Exploration/Chapter**: This is a structured learning experience that is part of a story, and provides the learner with an active way to learn new concepts, as well as targeted feedback. It is the core unit of learning in Oppia. The flow/screen that appears when any story is started is known as the Exploration/Chapter.
```

File: wiki/Terminology-in-Oppia.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+6. **Concept Card**: This is a non-story-based explanation of how to perform a particular skill. It serves as a reference/reminder for students who may have encountered the skill before but forgotten how to carry it out. These can access from "Revision" tab or are linked within the chapter you are playing.
```
Comment (adhiamboperes) [inline wiki/Terminology-in-Oppia.md]:
```suggestion
6. **Concept Card**: This is a non-story-based explanation of how to perform a particular skill. It serves as a reference/reminder for students who may have encountered the skill before but forgotten how to carry it out. These can be accessed from  the "Revision" tab or are linked within the chapter you are playing. 
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @MohitGupta121, I have left some additional grammar changes.

File: wiki/Terminology-in-Oppia.md (hunk 6 above)
Comment (MohitGupta121) [reply wiki/Terminology-in-Oppia.md]: Done.

File: wiki/Terminology-in-Oppia.md (hunk 7 above)
Comment (MohitGupta121) [reply wiki/Terminology-in-Oppia.md]: Done.

File: wiki/Terminology-in-Oppia.md (hunk 8 above)
Comment (MohitGupta121) [reply wiki/Terminology-in-Oppia.md]: Done.

File: wiki/Terminology-in-Oppia.md (hunk 9 above)
Comment (MohitGupta121) [reply wiki/Terminology-in-Oppia.md]: Done.

**Review by MohitGupta121 — COMMENTED** [review]
@adhiamboperes Thanks, PTAL.

**Review by adhiamboperes — APPROVED** [review]
Thanks @MohitGupta121, LGTM!

**Pull request merged**